modules/candidate_parameters/ajax/formHandler.php

```diff
@@ -477,12 +477,25 @@ function editConsentStatusFields($db, $user)
                           requires only the date of consent.');
                     return;
                 }
-            } else { // If no status stays no or record existed as NULL,
-                     // consent date and empty withdrawal date still required
+            } else { // If no status stays no or record existed as NULL
                 if (($oldStatus === null || $oldStatus === 'no') && !empty($date)
-                    && empty($withdrawal)
                 ) {
-                    $validated = true;
+                    $countYesStatusHistory = $db->pselectOne(
```

Comment:
okay so good catch @jesscall . i think the best way to do is to, instead of querying again, define $oldWithdrawal like how $oldStatus was by using candidate->getConsents() and check the new $withdrawal against those